iceoryx_posh/source/runtime/ipc_interface_base.cpp

```diff
@@ -77,12 +78,47 @@ std::string IpcMessageErrorTypeToString(const IpcMessageErrorType msg) noexcept
     return convert::toString(static_cast<UnderlyingType>(msg));
 }
 
+iox::optional<RuntimeName_t> ipcChannelNameToInterfaceName(RuntimeName_t channelName)
+{
+    RuntimeName_t interfaceName = iceoryxResourcePrefix(roudi::DEFAULT_UNIQUE_ROUDI_ID);
+    if (interfaceName.size() + channelName.size() > RuntimeName_t::capacity())
+    {
+        return nullopt;
+    }
+    interfaceName.append(TruncateToCapacity, channelName);
+    return interfaceName;
+}
+
 template <typename IpcChannelType>
 IpcInterface<IpcChannelType>::IpcInterface(const RuntimeName_t& runtimeName,
                                            const uint64_t maxMessages,
                                            const uint64_t messageSize) noexcept
-    : m_runtimeName(runtimeName)
 {
+    if (runtimeName.empty())
+    {
+        IOX_PANIC("Then runtime name must not be empty");
+    }
+    for (const auto s : platform::IOX_PATH_SEPARATORS)
```

Comment:
Same code like in the `roudi.cpp`. Maybe we should create a runtime name type right from the beginning. Maybe it is as easy like:
```cxx
class RuntimeName {
  FileName value;
  public:
    static iox::optional<RuntimeName> create(iox::string<FileName::capacity()> & value);
    const iox::string<FileName::capacity>& as_string() const { return value.as_string(); }
};
```

(But maybe I am mixing up iceoryx1 and 2)

Reply:
Yeah, it would definitely have helped if we had this from the beginning. It would be out of scope for this PR from my point of view.